stable_diffusion_videos/flax_stable_diffusion_pipeline.py

```diff
@@ -0,0 +1,1054 @@
+            audio_duration = num_step / fps
+
+            self.make_clip_frames(
+                params,
+                prompt_a,
+                prompt_b,
+                seed_a,
+                seed_b,
+                num_interpolation_steps=num_step,
+                save_path=save_path,
+                num_inference_steps=num_inference_steps,
+                guidance_scale=guidance_scale,
+                eta=eta,
+                height=height,
+                width=width,
+                upsample=upsample,
+                batch_size=batch_size,
+                T=get_timesteps_arr(
+                    audio_filepath,
+                    offset=audio_offset,
+                    duration=audio_duration,
+                    fps=fps,
+                    margin=margin,
+                    smooth=smooth,
+                )
+                if audio_filepath
+                else None,
+                skip=skip,
+                negative_prompt=negative_prompt,
+                jit=jit,
+            )
+            make_video_pyav(
+                save_path,
+                audio_filepath=audio_filepath,
+                fps=fps,
+                output_filepath=step_output_filepath,
+                glob_pattern=f"*{image_file_ext}",
+                audio_offset=audio_offset,
+                audio_duration=audio_duration,
+                sr=44100,
+            )
+
+        return make_video_pyav(
+            save_path_root,
+            audio_filepath=audio_filepath,
+            fps=fps,
+            audio_offset=audio_start_sec,
+            audio_duration=sum(num_interpolation_steps) / fps,
+            output_filepath=output_filepath,
+            glob_pattern=f"**/*{image_file_ext}",
+            sr=44100,
+        )
+
+    def embed_text(
+        self, params: Union[Dict, FrozenDict], text: str, negative_prompt=None
+    ):
+        """Helper to embed some text"""
+        prompt_ids = self.prepare_inputs(text)
+        embed = self.text_encoder(prompt_ids, params=params)[0]
+        return embed
+
+    def init_noise(self, prng_seed, noise_shape, dtype):
+        """Helper to initialize noise"""
+        noise = jax.random.normal(prng_seed, shape=noise_shape, dtype=dtype)
+        return noise
+
+    # TODO: what is this?
+    # @classmethod
+    # def from_pretrained(cls, *args, tiled=False, **kwargs):
+    #     """Same as diffusers `from_pretrained` but with tiled option, which makes images tilable"""
+    #     if tiled:
+
+    #         def patch_conv(**patch):
+    #             cls = nn.Conv2d
+    #             init = cls.__init__
+
+    #             def __init__(self, *args, **kwargs):
+    #                 return init(self, *args, **kwargs, **patch)
+
+    #             cls.__init__ = __init__
+
+    #         patch_conv(padding_mode="circular")
+
+    #     pipeline = super().from_pretrained(*args, **kwargs)
+    #     pipeline.tiled = tiled
+    #     return pipeline
```

Comment:
What is this doing? Should we adapt it for the flax version?

Reply:
If you set `tiled=True` in `from_pretrained`, it would patch Conv2d to use 'circular' convolutions which end up producing infinitely tillable generated images.

Fine if you leave this out if its not super simple. don't think the feature gets used too much

Reply:
I'll leave this for later :p